### PR TITLE
lib: stdbool header from the FreeBSD project

### PR DESCRIPTION
**Pull request by vwadekar**
This patch pulls the stdbool.h header file from the FreeBSD
project. The platforms require this header to fix many MISRA
defects among other things.

Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

### CODE REVIEW

File: include/lib/stdlib/stdbool.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ * $FreeBSD$
+ */
+
+/*
+ * Portions copyright (c) 2017, ARM Limited and Contributors.
+ * All rights reserved.
```
Comment (danh-arm) [inline include/lib/stdlib/stdbool.h]: Did you need to change anything from the FreeBSD version? It would be preferable if you just copied the FreeBSD version as-is with zero modifications. If you do need to change something, then please do that in a separate patch.

**Comment by vwadekar** [comment]
@danh-arm 

I have pushed the header without any modifications.

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by vwadekar** [comment]
@danh-arm 

Does this look OK? If yes, I can merge this commit to our downstream repo.
